### PR TITLE
Add design project templates

### PR DESCRIPTION
**Pull request by rrhyne**
Adds design project templates to be used in creating design issues and a link to them via the design process.

### CODE REVIEW

File: .idea/modules.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (danielmarquespt) [inline .idea/modules.xml]: Should these .idea files be part of the Pull Request?

**Review by danielmarquespt — APPROVED** [review]
Thanks for adding these to the handbook. The only comment I have is about the . files that are part of this PR

**Comment by danielmarquespt** [comment]
Also, should we add these here? https://github.com/sourcegraph/sourcegraph/tree/main/.github/ISSUE_TEMPLATE

It would be helpful, but at the same time it would create a lot of options that are only used by designers here https://github.com/sourcegraph/sourcegraph/issues/new/choose

File: .idea/modules.xml (hunk 1 above)
Comment (rrhyne) [reply .idea/modules.xml]: Thanks, they should not. I removed and added .idea to the .gitignore

**Comment by rrhyne** [comment]
> Also, should we add these here? https://github.com/sourcegraph/sourcegraph/tree/main/.github/ISSUE_TEMPLATE
> 
> It would be helpful, but at the same time it would create a lot of options that are only used by designers here https://github.com/sourcegraph/sourcegraph/issues/new/choose

I chose not to because many developers feel like we have too many issue templates.

File: content/departments/engineering/design/templates/extra-large-template.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  - These efforts require close collaboration with the Product Manager to scope and define the user problems being solved.
+
+✂️ -------------------------
+
+```markdown:
+## Design process
+
+  - Estimate: More than a month
+  - Weighting factors:
+    - Note: a project of this size requires resources that increase its risk profile.
```
Comment (jjinnii) [inline content/departments/engineering/design/templates/extra-large-template.md:19]: What does it mean to provide resources that increase a risk profile?

Comment (rrhyne) [reply content/departments/engineering/design/templates/extra-large-template.md:19]: More time and more people increase risk.

**Pull request merged**